File: android/java/org/chromium/chrome/browser/HuhiFeatureList.java
/* Copyright (c) 2020 The Huhi Software Authors. All rights reserved.
 * This Source Code Form is subject to the terms of the Huhi Software
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.chromium.chrome.browser;

import org.chromium.base.annotations.JNINamespace;
import org.chromium.base.annotations.NativeMethods;

@JNINamespace("chrome::android")
public abstract class HuhiFeatureList {
    public static final String HUHI_REWARDS = "HuhiRewards";
    public static final String USE_DEV_UPDATER_URL = "UseDevUpdaterUrl";
    public static final String FORCE_WEB_CONTENTS_DARK_MODE = "WebContentsForceDark";
    public static final String ENABLE_FORCE_DARK = "enable-force-dark";

    public static void enableFeature(String featureName, boolean enabled) {
        HuhiFeatureListJni.get().enableFeature(featureName, enabled);
    }

    @NativeMethods
    interface Natives {
        void enableFeature(String featureName, boolean enabled);
    }
}

File: browser/android/huhi_feature_list.cc
/* Copyright (c) 2020 The Huhi Authors. All rights reserved.
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 3.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

#include <string>

#include "base/android/jni_string.h"
#include "base/feature_list.h"
#include "huhi/browser/huhi_browser_process_impl.h"
#include "huhi/build/android/jni_headers/HuhiFeatureList_jni.h"
#include "chrome/browser/about_flags.h"
#include "components/flags_ui/pref_service_flags_storage.h"

namespace chrome {
namespace android {

static void JNI_HuhiFeatureList_EnableFeature(JNIEnv* env,
    const base::android::JavaParamRef<jstring>& featureName,
    jboolean enabled) {
  std::string feature_name = ConvertJavaStringToUTF8(env, featureName);
  enabled ? feature_name += "@1" : feature_name += "@0";
  flags_ui::PrefServiceFlagsStorage flags_storage(
      g_huhi_browser_process->local_state());
  about_flags::SetFeatureEntryEnabled(&flags_storage, feature_name, true);
}

}  // namespace android
}  // namespace chrome
